# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxUserFontSet.cpp
# Commit: 3a8c6048c5d0
# Full Hash: 3a8c6048c5d0874dd3d32c06e7923625b9383e8d
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2019-05-09 15:34:57
# Regressor Bug: 1545177
# File Overlap Count: 2
# Description:
#   Bug 1545177 - If descriptors of a FontFace are modified after creation, update the associated font entry so that face selection will respect the new values, and mark font sets as dirty. r=heycam
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27980
# ==============================================================================

diff -r c24c083425d0 -r 3a8c6048c5d0 gfx/thebes/gfxUserFontSet.cpp
--- a/gfx/thebes/gfxUserFontSet.cpp	Thu May 09 06:41:13 2019 +0000
+++ b/gfx/thebes/gfxUserFontSet.cpp	Thu May 09 09:43:42 2019 +0000
@@ -126,6 +126,27 @@
   mRangeFlags = aRangeFlags;
 }
 
+void gfxUserFontEntry::UpdateAttributes(
+    WeightRange aWeight, StretchRange aStretch, SlantStyleRange aStyle,
+    const nsTArray<gfxFontFeature>& aFeatureSettings,
+    const nsTArray<gfxFontVariation>& aVariationSettings,
+    uint32_t aLanguageOverride, gfxCharacterMap* aUnicodeRanges,
+    StyleFontDisplay aFontDisplay, RangeFlags aRangeFlags) {
+  // Remove the entry from the user font cache, if present there, as the cache
+  // key may no longer be correct with the new attributes.
+  gfxUserFontSet::UserFontCache::ForgetFont(this);
+
+  mFontDisplay = aFontDisplay;
+  mWeightRange = aWeight;
+  mStretchRange = aStretch;
+  mStyleRange = aStyle;
+  mFeatureSettings = aFeatureSettings;
+  mVariationSettings = aVariationSettings;
+  mLanguageOverride = aLanguageOverride;
+  mCharacterMap = aUnicodeRanges;
+  mRangeFlags = aRangeFlags;
+}
+
 gfxUserFontEntry::~gfxUserFontEntry() {
   // Assert that we don't drop any gfxUserFontEntry objects during a Servo
   // traversal, since PostTraversalTask objects can hold raw pointers to